src/erdstall.ts

```diff
@@ -62,33 +65,62 @@ export interface Leaver extends Exiter, Withdrawer {
 	leave(): Promise<Stages<Promise<ethers.ContractTransaction>>>;
 }
 
-export interface Subscriber {
+export interface BoundSubscriber {
 	subscribe(): Promise<void>;
+}
+
+export interface Subscriber {
+	subscribeAll():Promise<void>;
+	subscribeFor(who: Address): Promise<void>;
+}
+
+export interface Onboarder {
 	onboard(): Promise<void>;
 }
 
-export interface Erdstall
+export interface ErdstallClient
 	extends Watcher,
+		Subscriber {
+	readonly enclaveConn: EnclaveReader;
+	initialize(): Promise<void>;
+}
+
+export interface ErdstallSession
+	extends ErdstallClient,
+		BoundSubscriber,
 		Transactor,
 		Minter,
 		Trader,
 		Depositor,
 		Withdrawer,
 		Exiter,
-		Subscriber,
+		BoundSubscriber,
 		Leaver {
 	readonly address: Address;
 	initialize(): Promise<void>;
 }
 
 export function NewClient(
```

Comment:
Make alternative constructors static functions on the class they create. And it's convention to start all functions and methods with a small letter.